### PR TITLE
feat: add npm badge to readme

### LINKED ISSUE #730: 🚀 Feature: Add npm version badge to README.md (and template)
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I don't remember why I didn't add a README.md badge that includes the npm version:

https://shields.io/badges/endpoint-badge

```html
<a href="https://npmjs.com/package/create-typescript-app">
  <img alt="npm package version" src="https://img.shields.io/npm/v/create-typescript-app?color=21bb42">
</a>
```

<a href="https://npmjs.com/package/create-typescript-app">
  npm package version
</a>

### Additional Info

In addition to adding this in the repo's README.md, it'll need to be added to the templating used for setting up other repos. Search files for `readme` _(case insensitive)_ in their title to see where that's done.

### PR DESCRIPTION
**Pull request by jaas666**
## PR Checklist

- [X] Addresses an existing open issue: fixes #730 
- [X] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [X] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Added npm version badge to readme file. Also added it to the template that creates the file: `src/steps/writeReadme/generateTopContent.ts`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,8 @@
     </a>
 	<img alt="Style: Prettier 🧹" src="https://img.shields.io/badge/style-prettier_🧹-21bb42.svg" />
     <img alt="TypeScript: Strict 💪" src="https://img.shields.io/badge/typescript-strict_💪-21bb42.svg" />
+    <img alt="npm package version" src="https://img.shields.io/npm/v/create-typescript-app?color=21bb42" />
+
```
Comment (JoshuaKGoldberg) [inline README.md:28]:
```suggestion
```

Small thing, no need for a newline here I think.

File: src/steps/writeReadme/generateTopContent.ts [hunk 2]
```diff
@@ -1,83 +1,87 @@
 import { Options } from "../../shared/types.js";
 
 export function generateTopContent(options: Options, existingBadges: string[]) {
-	const remainingExistingBadges = new Set(existingBadges);
-	const badges: string[] = [];
+  const remainingExistingBadges = new Set(existingBadges);
```
Comment (JoshuaKGoldberg) [inline src/steps/writeReadme/generateTopContent.ts:4]: Same as https://github.com/JoshuaKGoldberg/create-typescript-app/pull/881#pullrequestreview-1640908590 - looks like formatting is being applied differently?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Similar to #881: functionally looks great, thanks!

File: src/steps/writeReadme/generateTopContent.test.ts
Comment (JoshuaKGoldberg) [inline src/steps/writeReadme/generateTopContent.test.ts:1]: Looking at the test failures mentioned in https://github.com/JoshuaKGoldberg/create-typescript-app/actions/runs/6280600786/job/17065272770?pr=880, I think you'll want to run `pnpm run test -u` locally to update other test snapshots. https://vitest.dev/guide/mocking.html

File: README.md (hunk 1 above)
Comment (jaas666) [reply README.md]: Fixed

File: src/steps/writeReadme/generateTopContent.ts (hunk 2 above)
Comment (jaas666) [reply src/steps/writeReadme/generateTopContent.ts]: Should be fixed

File: src/steps/writeReadme/generateTopContent.test.ts
Comment (jaas666) [reply src/steps/writeReadme/generateTopContent.test.ts:1]: I think I fixed this now.

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 thanks!

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Hmm, actually, it looks like there are a few more test failures. You'll want to `pnpm run test -u` locally to update snapshots. LMK if that gives you any issues?

**Comment by jaas666** [comment]
Yikes. I think I failed to include one updated file. How about now?

All test are passing locally

**Review by JoshuaKGoldberg — APPROVED** [review]
That did it! I did the merge from `main` and resolved merge conflicts by just re-applying the same changes you did in this PR. 💯  Thanks!

**Pull request merged**